app/controlplane/internal/biz/robotaccount_integration_test.go

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package biz_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/chainloop-dev/chainloop/app/controlplane/internal/biz"
+	"github.com/chainloop-dev/chainloop/app/controlplane/internal/biz/testhelpers"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+func (s *robotAccountTestSuite) TestRevoke() {
+	ctx := context.Background()
+	s.Run("returns an error if org ID format is not valid", func() {
+		err := s.RobotAccount.Revoke(ctx, "not_valid_uuid", "52597CC8-21BF-45C4-B30B-AE2879556CE1")
+		s.ErrorAs(err, &biz.ErrInvalidUUID{})
+	})
+
+	s.Run("returns an error if robot account ID format is not valid", func() {
+		err := s.RobotAccount.Revoke(ctx, "991B156A-7636-44D2-97B0-E97070E626DC", "not_valid_uuid")
+		s.ErrorAs(err, &biz.ErrInvalidUUID{})
+	})
+
+	s.Run("returns a Not Found if robot account cannot be found", func() {
+		err := s.RobotAccount.Revoke(ctx, s.org.ID, "DCF3895A-22BE-49D0-99D4-0A76835D0BD1")
+		s.ErrorAs(err, &biz.ErrNotFound{})
+	})
+
+	s.Run("revokes the robot account", func() {
+		err := s.RobotAccount.Revoke(ctx, s.org.ID, s.ra.ID.String())
+		s.NoError(err)
+
+		// Reload the robot account
+		ra, err := s.RobotAccount.FindByID(ctx, s.ra.ID.String())
+		s.NoError(err)
+		s.NotNil(ra.RevokedAt)
+	})
+}
+
+// Utility struct to hold the test suite
+type robotAccountTestSuite struct {
+	testhelpers.UseCasesEachTestSuite
+	org *biz.Organization
+	ra  *biz.RobotAccount
+}
+
+// // Run the tests
+func TestRobotAccountUseCase(t *testing.T) {
+	suite.Run(t, new(robotAccountTestSuite))
+}
+
+func (s *robotAccountTestSuite) SetupTest() {
```

Comment:
btw, smth that I learnt not long time ago, and I know that's not consistent in other places, but you can instead of using `assert := assert.New(s.T())` and then assert.noError, you can do directly `s.NoError`)

So basically you can remove `assert` variable and instead use `s.NoError(err)`

You can probably also remove `t := s.T()` at that point too.

Reply:
Thanks, I was just blindly copying from elsewhere. I've simplified it a little bit.